src/quick-lint-js/parse.h

```diff
@@ -2516,7 +2516,18 @@ class parser {
 
     if (this->peek().type == token_type::kw_else) {
       this->skip();
+
+      bool semicolon_after_else = this->peek().type == token_type::semicolon;
+      const char8 *token_after_else_begin = this->peek().begin;
+
       parse_and_visit_body();
+      if (semicolon_after_else &&
+          this->peek().type == token_type::left_curly &&
+          !this->peek().has_leading_newline) {
+        this->error_reporter_->report(error_unexpected_semicolon_after_else{
+            .semicolon = source_code_span(token_after_else_begin, token_after_else_begin+1)
+        });
```

Comment:
Nit: If we store the whole token, I think the code is more readable:
```suggestion
      token token_after_else = this->peek();

      parse_and_visit_body();
      if (token_after_else.type == token_type::semicolon &&
          this->peek().type == token_type::left_curly &&
          !this->peek().has_leading_newline) {
        this->error_reporter_->report(error_unexpected_semicolon_after_else{
            .semicolon = source_code_span(token_after_else.begin, token_after_else.end)
        });
```

Nit: I'd rather store the token's `end` rather than assuming a semicolon ends 1 after its begin. (Suggested diff above incorporates this change.)